trajectory_follower_wrapper/src/trajectory_follower_wrapper_node.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2024 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "trajectory_follower_wrapper/trajectory_follower_wrapper_node.hpp"
+
+namespace trajectory_follower_wrapper
+{
+  namespace std_ph = std::placeholders;
+
+  TrajectoryFollowerWrapperNode::TrajectoryFollowerWrapperNode(const rclcpp::NodeOptions &options)
+      : carma_guidance_plugins::ControlPlugin(options)
+  {
+    // Create initial config
+    config_ = TrajectoryFollowerWrapperConfig();
+    config_.vehicle_response_lag = declare_parameter<double>("vehicle_response_lag", config_.vehicle_response_lag);
+    config_.time_threshold = declare_parameter<double>("time_threshold", config_.time_threshold);
+
+  }
+
+  rcl_interfaces::msg::SetParametersResult TrajectoryFollowerWrapperNode::parameter_update_callback(const std::vector<rclcpp::Parameter> &parameters)
+  {
+    auto error_double = update_params<double>({
+      {"vehicle_response_lag", config_.vehicle_response_lag},
+      {"time_threshold", config_.time_threshold}
+      }, parameters);
+
+    rcl_interfaces::msg::SetParametersResult result;
+
+    result.successful = !error_double;
+
+    return result;
+  }
+
+  carma_ros2_utils::CallbackReturn TrajectoryFollowerWrapperNode::on_configure_plugin()
+  {
+    // Reset config
+    config_ = TrajectoryFollowerWrapperConfig();
+
+    // Load parameters
+    get_parameter<double>("vehicle_response_lag", config_.vehicle_response_lag);
+    get_parameter<double>("time_threshold", config_.time_threshold);
+
+    RCLCPP_INFO_STREAM(rclcpp::get_logger("trajectory_follower_wrapper"), "Loaded Params: " << config_);
+    // Register runtime parameter update callback
+    add_on_set_parameters_callback(std::bind(&TrajectoryFollowerWrapperNode::parameter_update_callback, this, std_ph::_1));
+
+    // Setup subscribers
+    control_cmd_sub_ = create_subscription<autoware_auto_msgs::msg::AckermannControlCommand>("output/control_cmd", 10,
```

Comment:
This subscriber needs to be transient_local 
https://github.com/usdot-fhwa-stol/autoware.auto/blob/3450f94fa694f51b00de272d412722d65a2c2d3e/AutowareAuto/src/control/trajectory_follower_nodes/src/latlon_muxer_node.cpp#L33